checkbox-ng/plainbox/impl/test_new_resource.py

```diff
@@ -0,0 +1,494 @@
+        self.assertEqual(evaluated, result)
+
+        evaluated = all(evaluate_lazy(expr, namespace).values())
+        self.assertEqual(evaluated, result_bool)
+
+    def test_or_false(self):
+        expr = "namespace.b == 20 or namespace.a == 11"
+        namespace = {"namespace": [HD({"a": 1, "b": 2}), HD({"a": 2, "b": 2})]}
+        result = {"namespace": []}
+        result_bool = False
+
+        evaluated = evaluate(expr, namespace, explain_callback=print)
+        self.assertEqual(evaluated, result)
+
+        evaluated = all(evaluate_lazy(expr, namespace).values())
+        self.assertEqual(evaluated, result_bool)
+
+    def test_gt_true(self):
+        expr = "namespace.a > 1"
+        namespace = {"namespace": [HD({"a": 1, "b": 2}), HD({"a": 2, "b": 2})]}
+        result = {"namespace": [HD({"a": 2, "b": 2})]}
+        result_bool = True
+
+        evaluated = evaluate(expr, namespace, explain_callback=print)
+        self.assertEqual(evaluated, result)
+
+        evaluated = all(evaluate_lazy(expr, namespace).values())
+        self.assertEqual(evaluated, result_bool)
+
+    def test_gt_false(self):
+        expr = "namespace.a > 10"
+        namespace = {"namespace": [HD({"a": 1, "b": 2}), HD({"a": 2, "b": 2})]}
+        result = {"namespace": []}
+        result_bool = False
+
+        evaluated = evaluate(expr, namespace, explain_callback=print)
+        self.assertEqual(evaluated, result)
+
+        evaluated = all(evaluate_lazy(expr, namespace).values())
+        self.assertEqual(evaluated, result_bool)
+
+    def test_gte(self):
+        expr = "namespace.a >= 1"
+        namespace = {
+            "namespace": [HD({"a": 1, "b": 2}), HD({"a": 2, "b": 2})],
+        }
+        result = {"namespace": [HD({"a": 1, "b": 2}), HD({"a": 2, "b": 2})]}
+
+        evaluated = evaluate(expr, namespace, explain_callback=print)
+        self.assertEqual(evaluated, result)
+
+    def test_cast_int(self):
+        expr = "int(namespace.a) == 1"
+        namespace = {
+            "namespace": [HD({"a": "1", "b": "2"}), HD({"a": "2", "b": "2"})],
+        }
+        result = {"namespace": [HD({"a": "1", "b": "2"})]}
+
+        evaluated = evaluate(expr, namespace, explain_callback=print)
+        self.assertEqual(evaluated, result)
+
+    def test_cast_float(self):
+        expr = "float(namespace.a) == 1"
+        namespace = {
+            "namespace": [HD({"a": "1", "b": "2"}), HD({"a": "2", "b": "2"})],
+        }
+        result = {"namespace": [HD({"a": "1", "b": "2"})]}
+
+        evaluated = evaluate(expr, namespace, explain_callback=print)
+        self.assertEqual(evaluated, result)
+
+    def test_in(self):
+        expr = "namespace.a in ['1', '2']"
+        namespace = {
+            "namespace": [HD({"a": "1"}), HD({"a": "2"}), HD({"a": "3"})]
+        }
+        result = {"namespace": [HD({"a": "1"}), HD({"a": "2"})]}
+
+        evaluated = evaluate(expr, namespace)
+        self.assertEqual(evaluated, result)
+
```

Comment:
Add test for "not in". You would have caught the bug in that operator.
```suggestion
    def test_not_in(self):
        expr = "namespace.a not in ['1', '2']"
        namespace = {
            "namespace": [HD({"a": "1"}), HD({"a": "2"}), HD({"a": "3"})]
        }
        result = {"namespace": [HD({"a": "3"})]}

        evaluated = evaluate(expr, namespace)
        self.assertEqual(evaluated, result)

```

Reply:
good idea